### PR TITLE
ON-14979: Added a simple Github action to run unit tests

### PR DESCRIPTION
**Pull request by krishd-amd**
https://github.com/krishd-amd/kubernetes-onload/actions/runs/6261028520
https://github.com/krishd-amd/kubernetes-onload/actions/runs/6261028520/job/17000252527

A simple action to just run the golang unit tests. In the absence of a jenkins instance, this seems rather simple and easy to get it repeatedly running on commit.

### CODE REVIEW

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+jobs:
+  ut:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '>=1.17.0'
+      - run: make test
```
Comment (tcrawley-xilinx) [inline .github/workflows/unit-test.yml:12]:
```suggestion
      - run: make test

```
Just so that git doesn't complain

**Review by tcrawley-xilinx — APPROVED** [review]
Looks good!
Have you verified the failure case?

File: .github/workflows/unit-test.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+jobs:
+  ut:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '>=1.17.0'
```
[the remark this answers is not in the document]
Comment (ivatet-amd) [reply .github/workflows/unit-test.yml:11]: Updated [ON-15183](https://jira.xilinx.com/browse/ON-15183) to mention the minimum Golang version in README.md.

**Comment by krishd-amd** [comment]
> Looks good! Have you verified the failure case?

(https://github.com/krishd-amd/kubernetes-onload/actions/runs/6272815447/job/17035086741)

**Pull request merged**